android/java/org/chromium/chrome/browser/settings/BraveNewsPreferencesDetails.java

```diff
@@ -224,6 +227,16 @@ public void subscribeToNewDirectFeed(int position, Url feedUrl, boolean isFromFe
         });
     }
 
+    @Override
+    public void updateFeedSearchResultItem(int position, String url, String publisherId) {
+        if (!mFeedSearchResultItemFollowMap.containsKey(url)) {
+            mFeedSearchResultItemFollowMap.put(url, publisherId);
+        } else if (mFeedSearchResultItemFollowMap.containsKey(url)) {
```

Comment:
we don't need a second check on `mFeedSearchResultItemFollowMap.containsKey(url)`